### PR TITLE
Feat: Included Signed api for s3

### PR DESCRIPTION
**Pull request by kavinpanneer**
## Checklist

- [ ] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingsdk/src/std/sample.json [hunk 1]
```diff
@@ -0,0 +1,86 @@
+{
```
Comment (tsuf239) [inline libs/wingsdk/src/std/sample.json:1]: what is that for?

File: libs/wingsdk/src/target-sim/bucket.inflight.ts [hunk 2]
```diff
@@ -208,6 +208,32 @@ export class Bucket implements IBucketClient, ISimulatorResourceInstance {
...
+    /**
+     * Generating a mock signed url response
+     */
+    const expiresInSeconds= duration?.seconds||(24*3600);
+    const params= new URLSearchParams();
+    params.set("Bucket","my_bucket");
+    params.set("key",key);
+    params.set("Expires",String(expiresInSeconds));
+    params.set("SignatureVersion","v4");
+    const url= new URL("",undefined);
```
Comment (tsuf239) [inline libs/wingsdk/src/target-sim/bucket.inflight.ts:229]: can you use the `publicUrl` as the base? this way we'll get error handling for free :)

File: libs/wingsdk/src/target-sim/bucket.inflight.ts [hunk 3]
```diff
@@ -208,6 +208,32 @@ export class Bucket implements IBucketClient, ISimulatorResourceInstance {
...
+        ); 
+    }
+    /**
+     * Generating a mock signed url response
+     */
+    const expiresInSeconds= duration?.seconds||(24*3600);
+    const params= new URLSearchParams();
+    params.set("Bucket","my_bucket");
+    params.set("key",key);
+    params.set("Expires",String(expiresInSeconds));
```
Comment (tsuf239) [inline libs/wingsdk/src/target-sim/bucket.inflight.ts:227]: how do we know when something is truly expiring? potentially we can have a URL of 2 minutes, and retrieve it 3 minutes later when nothing stops us. 
When designing the expiration, please take in mind that we initiate the class within each compilation...

File: libs/wingsdk/src/target-sim/bucket.inflight.ts [hunk 4]
```diff
@@ -208,6 +208,32 @@ export class Bucket implements IBucketClient, ISimulatorResourceInstance {
...
+    params.set("SignatureVersion","v4");
```
Comment (tsuf239) [inline libs/wingsdk/src/target-sim/bucket.inflight.ts:228]: what is that for?

**Comment by tsuf239** [comment]
also, you need to apply the new method `signedUrl` to the interface export interface `IBucketClient`, and to `bucket._addInflightOps` and to the permissions file of shared-aws: `libs/wingsdk/src/shared-aws/permissions.ts`

File: libs/wingsdk/src/shared-aws/bucket.inflight.ts [hunk 5]
```diff
@@ -254,17 +254,23 @@ export class BucketClient implements IBucketClient {
...
    * access the file until the link expires (defaults to 24 hours).
    * @param key The key to reach
    * @param duration Time until expires
    */
   public async signed_url(key: string, duration?: Duration): Promise<string> {
-    // for signed_url take a look here: https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/s3-example-creating-buckets.html#s3-create-presigendurl-get
-    throw new Error(
-      `signed_url is not implemented yet (key=${key}, duration=${duration})`
-    );
+    const expiryTime:number= duration?.hours|| 24;
```
Comment (tsuf239) [inline libs/wingsdk/src/shared-aws/bucket.inflight.ts:266]: that's isn't hours, but seconds: 
image

**Comment by hasanaburayyan** [comment]
Just a heads up on the pr title you will want to follow these conventions: https://www.winglang.io/contributing/start-here/pull_requests

Basically the title will need to be something like `feat(sdk): description`

**Pull request closed without merging**